apps/schedule/schedule_json.py

```diff
@@ -0,0 +1,128 @@
+            "start": event_start().strftime("%Y-%m-%d"),
+            "time_zone_name": event_tz,
+            "timeslot_duration": "00:10",
+            "title": "Electromagnetic Field {}".format(event_year()),
+            "url": external_url("main"),
+        },
+    }
+
+    for day in events_per_day_and_room:
+        day_schedule = {
+            "date": day["start"].strftime("%Y-%m-%d"),
+            "day_end": day["start"].isoformat(),
+            "day_start": day["end"].isoformat(),
+            "index": day["index"],
+            "rooms": {},
+        }
+        for room, events in sorted(day["rooms"].items()):
+            day_schedule["rooms"][room] = []
+            for proposal in events:
+                links = {
+                    proposal.c3voc_url,
+                    proposal.youtube_url,
+                    proposal.thumbnail_url,
+                    proposal.map_link,
+                }
+                links.discard(None)
+                links.discard("")
+                day_schedule["rooms"][room].append(
+                    {
+                        "abstract": None,  # The proposal model does not implement abstracts
+                        "attachments": [],
+                        "date": event_tz.localize(proposal.start_date).isoformat(),
+                        "description": proposal.description,
+                        "do_not_record": proposal.video_privacy != "public",
+                        "duration": get_duration(proposal.start_date, proposal.end_date),
+                        "guid": None,
+                        "id": proposal.id,
+                        # This assumes there will never be a non-english talk,
+                        # which is probably fine for a conference in the UK.
+                        "language": "en",
+                        "links": sorted(links),
+                        "persons": [
+                            {
+                                "name": name.strip(),
+                                "public_name": name.strip(),
+                            }
+                            for name in (proposal.published_names or proposal.user.name).split(",")
+                        ],
+                        "recording_license": "CC BY-SA 3.0",
+                        "room": room,
+                        "slug": "emf{}-{}-{}".format(
+                            event_year(),
+                            proposal.id,
+                            proposal.slug,
+                        ),
+                        "start": event_tz.localize(proposal.start_date).strftime("%H:%M"),
+                        "subtitle": None,
+                        "title": proposal.display_title,
+                        # Contrary to the infobeamer frab module, the json module does not allow users to set colours
+                        # for tracks themselves. It instead relies on the schedule itself to provide those colours.
+                        "track": None,
```

Comment:
We could go ahead and hard-code a mapping `proposal.type` to colour, but i'm not sure if that's a good idea

Reply:
I think we can leave it for now